app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt

```diff
@@ -197,7 +218,16 @@ class CalendarFragment :
     override fun onOptionsItemSelected(item: MenuItem): Boolean {
         when (item.itemId) {
             R.id.action_switch_view_mode -> {
-                isWeekMode = !isWeekMode
+                if (isWeekMode) {
+                    isWeekMode = !isWeekMode
+                }
+                else if (isMonthMode) {
+                    isMonthMode = !isMonthMode
+                    isWeekMode = !isWeekMode
+                }
+                else {
+                    isMonthMode = !isMonthMode
+                }
```

Comment:
This seems very much like booleans are not the right way to model this state. 
Could we migrate this to an `enum`?

Reply:
```
private enum class ViewMode {
         DAY, WEEK, MONTH
}

private var viewMode: ViewMode = ViewMode.MONTH
```

```suggestion
  viewMode = when (viewMode) {
                    ViewMode.DAY -> ViewMode.MONTH
                    ViewMode.WEEK -> ViewMode.DAY
                    ViewMode.MONTH -> ViewMode.WEEK
                }              
```